src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -108,6 +108,26 @@ public static async Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout)
             return task.GetAwaiter().GetResult();
         }
 
+        /// <summary>
+        /// Gets the <see cref="Task{TResult}.Result"/> of a completed <see cref="Task{TResult}"/>.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the result produced by the <see cref="Task{TResult}"/>.</typeparam>
+        /// <param name="task">The completed task.</param>
+        /// <returns>The result value of the <see cref="Task{TResult}"/>.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="task"/> is <see langword="null"/>.</exception>
+        /// <exception cref="InvalidOperationException">If <paramref name="task"/> is not completed.</exception>
+        public static TResult CompletedResult<TResult>(this Task<TResult> task)
```

Comment:
Should we also have a method for non-generic `Task<T>` for folks that just want to rethrow any exceptions on a completed Task?

Reply:
In a project where I implemented this previously, IntelliSense was not as friendly when both methods were included. It would be totally reasonable to say this should be treated as a Roslyn bug and not influence the API design for an unrelated library.